drs_cli/client.py

```diff
@@ -30,18 +33,27 @@ class DRSClient():
     def __init__(
         self,
         host: str = 'http://0.0.0.0',
-        port: int = '8080',
+        port: int = 80,
         base_path: str = 'ga4gh/drs/v1',
+        drs_uri: Optional[str] = None,
         token: Optional[str] = None,
-    ) -> None:
+    ) -> Union[None, Error]:
+        
         self.url = f"{host}:{port}/{base_path}"
+        if drs_uri:
+            if _check_drs_regex(drs_uri):
+                drs_host = _get_host_and_id(drs_uri)[0]
+                self.url = f"http://{drs_host}/{base_path}"
```

Comment:
I think you can refactor the code to assign to `self.uri` only once, at the end. I would also cover for the port problem when passing a DRS URI: I would set the default port to 443 instead of 80. Then, when the user passes a a HTTP (and not HTTPS) URL but not a port, then change it to 80. But if the user supplies a port, then use the provided port, even for DRS URIs. This is fine as we are not breaking the specs, the DRS URI does not contain the port, we provide it manually. In this way we can support the case where we don't have HTTPS DRS instances during development and/or are deploying to different ports.